### ISSUE TITLE
Solution soundness issue on QF_BV formula

### ISSUE THREAD

**Issue by rainoftime** [SRI-CSL/yices2#306]
Hi, for the following formula

yices 5019920 yields sat, but z3 and cvc4 return unsat
~~~~
(set-logic QF_BV)
(declare-const x (_ BitVec 8))
(declare-const y (_ BitVec 8))
(declare-const z (_ BitVec 8))
(assert (or (not (= (bvadd x y z) (bvadd (bvadd x y) z)))
            (not (= (bvmul x y z) (bvmul (bvmul x y) z)))
            (not (= (bvand x y z) (bvand (bvand x y) z)))
            (not (= (bvor x y z) (bvor (bvor x y) z)))
            (not (= (bvxor x y z) (bvxor (bvxor x y) z)))
            (not (= (bvxnor x y z) (bvxnor (bvxnor x y) z)))))
(check-sat)
~~~~

**Comment by rainoftime** [comment]
boolector developer comment: Boolector/boolector#150 (comment)

**Comment by aniemetz** [comment]
As mentioned in the Boolector issue referenced above, operator XNOR was mistakenly marked as :left-associative in SMT-LIB, this has been corrected recently. The behavior for n-ary XNORs is not defined in SMT-LIB now.

**Comment by aniemetz** [comment]
CVC4 previously still treated bvxnor as left-associative, fixed here: CVC4/CVC4#5138

**Comment by BrunoDutertre** [comment]
Disallowed n-ary bvxnor (087ceb67debe42e651ad6a3d5c483c94c5db7e98).
